test/FunctionalTests/Microsoft.OData.Client.Tests/ALinq/DollarApplyGroupByTests.cs

```diff
@@ -0,0 +1,1717 @@
+                "Product/TaxRate with min as MinTaxRate,Product/TaxRate with max as MaxTaxRate))", serviceUri),
+                aggregateQuery.ToString());
+
+            MockGroupByMultipleNavProperties_AggregationsTargetingNavProperty();
+
+            var aggregateResult = aggregateQuery.ToArray();
+
+            Assert.Equal(2, aggregateResult.Length);
+        }
+
+        [Fact]
+        public void GroupByResultSelector_ByConstant_TargetingNavProperty()
+        {
+            var queryable = this.dsContext.CreateQuery<Sale>(salesEntitySetName);
+
+            var aggregateQuery = queryable.GroupBy(d1 => 1,
+                (d1, d2) => new
+                {
+                    SumTaxRate = d2.Sum(d3 => d3.Product.TaxRate),
+                    AvgTaxRate = d2.Average(d3 => d3.Product.TaxRate),
+                    MinTaxRate = d2.Min(d3 => d3.Product.TaxRate),
+                    MaxTaxRate = d2.Max(d3 => d3.Product.TaxRate)
+                });
+
+            Assert.Equal(
+                string.Format("{0}/Sales?$apply=aggregate(" +
+                "Product/TaxRate with sum as SumTaxRate,Product/TaxRate with average as AvgTaxRate," +
+                "Product/TaxRate with min as MinTaxRate,Product/TaxRate with max as MaxTaxRate)", serviceUri),
+                aggregateQuery.ToString());
+
+            MockGroupByConstant_AggregationsTargetingNavProperty();
+
+            var aggregateResult = aggregateQuery.ToArray();
+
+            Assert.Single(aggregateResult);
+        }
+
+        [Fact]
+        public void GroupByResultSelector_ByConstant_MixedScenarios()
+        {
+            var queryable = this.dsContext.CreateQuery<Sale>(salesEntitySetName);
+
+            var aggregateQuery = queryable.GroupBy(d1 => 1,
+                (d1, d2) => new
+                {
+                    GroupingConstant = d1,
+                    GibberishConstant = "dfksjfl",
+                    SumTaxRate = d2.Sum(d3 => d3.Product.TaxRate),
+                    AvgAmount = d2.Average(d3 => d3.Amount),
+                    MinTaxRate = d2.Min(d3 => d3.Product.TaxRate),
+                    MaxAmount = d2.Max(d3 => d3.Amount),
+                    GroupCount = d2.Count(),
+                    DistinctCurrency = d2.CountDistinct(d3 => d3.Currency.Code)
+                });
+
+            Assert.Equal(
+                string.Format("{0}/Sales?$apply=aggregate(" +
+                "Product/TaxRate with sum as SumTaxRate,Amount with average as AvgAmount," +
+                "Product/TaxRate with min as MinTaxRate,Amount with max as MaxAmount," +
+                "$count as GroupCount,Currency/Code with countdistinct as DistinctCurrency)", serviceUri),
+                aggregateQuery.ToString());
+
+            MockGroupByConstant_MixedScenarios();
+
+            var aggregateResult = aggregateQuery.ToArray();
+
+            Assert.Single(aggregateResult);

```

Comment:
nit: consider adding assertions to verify the properties are set, including what the expected value of `GibberishConstant` is in the result.

Reply:
Done for this and other tests